File: pset6/credit/credit.py
from cs50 import get_int
sums = []
sum_odd = 0

# Prompting the user for number
number = get_int("Number: ")

# Storing the number in a list
num = [int(x) for x in str(number)]

# Accessing the number's second-to-last digits
for i in range(len(num)-2, -1, -2):
    # Multiplying each digit by 2
    tmp = [int(y) for y in str(num[i] * 2)]
    # Storing the digits of the product result in a list
    sums[len(sums):] = tmp

# Accessing the other digits that wren't multiplied then sums them
for j in range(len(num)-1, -1, -2):
    sum_odd += num[j]

# Checking the total's last digit if 0 or not
if ((sum_odd + sum(sums)) % 10 != 0):
    print("INVALID")
# If the check is corect, determine the type of the card
else:
    ident = str(num[0]) + str(num[1])
    if len(num) == 15 and (ident == '34' or ident == '37'):
        print('AMEX')
    elif len(num) == 16 and (ident == '51' or ident == '52' or ident == '53' or ident == '54' or ident == '55'):
        print('MASTERCARD')
    elif (len(num) == 13 or len(num) == 16) and (ident[0] == '4'):
        print("VISA")
    else:
        print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int j, k = 0, l = 0 ;
    long n = get_long("number=");
    long m = n;
    for (int i = 1; n != 0; i++)
    {
        // j is the digit extracted from the number starting from right to the left 
        j = n % 10;
        n = (n - (n % 10)) / 10;
        if (i % 2 != 0)
        {
            // k is the sum of the odd digits of the entered number starting from the most right digit
            k = k + j;  
        }
        else 
        {
            // l is the sum of the even digits of the entered number starting from the next to most right digit (after multiplied by 2)
            if (2 * j < 9)
            {
                l = l + 2 * j; 
            }
            else 
            {
                l = l + ((2 * j / 10) + (2 * j % 10));
            }
        }
    }
    if (((l + k) % 10) != 0)
    {
        printf("INVALID\n");
    }
    else 
    {
        if (((m / 1000000000000) == 4) || ((m / 1000000000000000) == 4))
        {
            printf("VISA\n");
        }
        else if ((m / 10000000000000) == 34 || (m / 10000000000000) == 37)
        {
            printf("AMEX\n");
        }
        else if ((m / 100000000000000) == 51 || (m / 100000000000000) == 52 || (m / 100000000000000) == 53 || (m / 100000000000000) == 54
                 || (m / 100000000000000) == 55)
        {
            printf("MASTERCARD\n");
        }
        else 
        {
            printf("INVALID\n");
        }
        
    }
}